clients/mongoStoreClient.go

```diff
@@ -232,7 +231,7 @@ func (d MongoStoreClient) ExecuteQuery(details *model.QueryData) ([]byte, error)
 
 	err := sessionCopy.DB("").C(DEVICE_DATA_COLLECTION).
 		Find(query).
-		Sort(sort_time_descending). //we will always sort by time
+		Sort(query_fields...). //sort by time but use full index
```

Comment:
I think the Sort field needs to be `uploadid_query_fields` if the query includes an uploadIds clause (otherwise I'm not sure if it will use the index at all for the find and/or sorting). So, we might need some sort of switch here.  Or, an alternative would be to push a hint to mongo for it to always use the `uploadid_query_fields` for the find as well as the sort. I'm not positive, but I think this might work. We'd have to test to ensure that forcing it to use the larger index is okay (it actually uses the index).  If it is, I wonder if we can drop the `query_fields` index thereby saving memory/disk and increasing overall performance (since every index you need to keep up-to-date does slightly slow performance).

Thoughts?